go/cmd/dolt/commands/diff_output.go

```diff
@@ -390,9 +450,138 @@ func (j *jsonDiffWriter) RowWriter(ctx context.Context, td diff.TableDelta, unio
 	return j.rowDiffWriter, err
 }
 
+func (j *jsonDiffWriter) WriteTriggerDiff(ctx context.Context, triggerName, oldDefn, newDefn string) error {
+	if j.triggersWritten == 0 {
+		// begin the document if necessary
+		if j.tablesWritten == 0 {
+			_, err := j.wr.Write([]byte("{"))
+			if err != nil {
+				return err
+			}
+		} else {
+			_, err := j.wr.Write([]byte(jsonDataDiffFooter + ","))
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if j.triggersWritten == 0 {
+		_, err := j.wr.Write([]byte(`"triggers":[`))
+		if err != nil {
+			return err
+		}
+	} else {
+		_, err := j.wr.Write([]byte(","))
+		if err != nil {
+			return err
+		}
+	}
+
+	triggerNameBytes, err := ejson.Marshal(triggerName)
+	if err != nil {
+		return err
+	}
+
+	oldDefnBytes, err := ejson.Marshal(oldDefn)
+	if err != nil {
+		return err
+	}
+
+	newDefnBytes, err := ejson.Marshal(newDefn)
+	if err != nil {
+		return err
+	}
+
+	_, err = j.wr.Write([]byte(fmt.Sprintf(`{"name":%s,"from_definition":%s,"to_definition":%s}`,
+		triggerNameBytes, oldDefnBytes, newDefnBytes)))
+	if err != nil {
+		return err
+	}
+
+	j.triggersWritten++
+	return nil
+}
+
+func (j *jsonDiffWriter) WriteViewDiff(ctx context.Context, viewName, oldDefn, newDefn string) error {
+	if j.viewsWritten == 0 {
+		// begin the document if necessary
+		if j.tablesWritten == 0 && j.triggersWritten == 0 {
```

Comment:
This seems frail. I'd add a method which adds the "{" char once and only once. Then you can call it whenever you think you may need to start the document.

Reply:
Introduced a method for this logic.